### PR TITLE
Profiler Plugin

### PR DESCRIPTION
**Pull request by kolanos**
This plugin uses Python's `cProfile` module to profile an invocation.

### CODE REVIEW

**Comment by ewindisch** [comment]
Awesome!
I believe this should live in its own repo though!
We'll need to figure out the upstream signing api changes still.

File: iopipe/contrib/profiler/plugin.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            self.stats = pstats.Stats(self.profile, stream=self.stream)
+            self.stats.sort_stats(*self.sort)
+            self.stats.print_stats(*self.restrictions)
+
+            signed_request = get_signed_request(report)
+            upload_profiler_report(signed_request['signedRequest'], self.stream)
+
+            plugin = [p for p in report.report['plugins'] if p['name'] == self.name]
+            if plugin:
+                plugin[0]['uploads'] = [signed_request['url']]
```
Comment (kolanos) [inline iopipe/contrib/profiler/plugin.py]: @pselle Is this how we add the upload to the report? Not seeing anything like this in the node.js profiler.

Comment (pselle) [reply iopipe/contrib/profiler/plugin.py]: @kolanos will respond to this in Slack

File: README.md [hunk 2]
```diff
@@ -129,6 +130,61 @@ This makes it easy to add custom data and telemetry within your function.
...
+from iopipe.contrib.profiler import ProfilerPlugin
+
+iopipe = IOpipe(plugins=[ProfilerPlugin()])
+
+@iopipe
+def handler(event, context):
+    # do something here
+```
+
+By default the plugin will be disabled and can be enabled at runtime by setting the `IOPIPE_ENABLE_PROFILER` environment variable to `true`/`True`.
```
Comment (pselle) [inline README.md]: Let's move to the `IOPIPE_PROFILER*` namespace (`IOPIPE_PROFILER_ENABLE`)

File: iopipe/contrib/profiler/plugin.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    name = 'profiler'
+    version = '0.1.0'
+    homepage = 'https://github.com/iopipe/iopipe-python'
+
+    def __init__(self, enabled=False, sort='cumulative', restrictions=10):
+        """
+        Instantiates the profiler plugin
+
+        :param enabled: Whether or not the profiler should be enabled for all invocations.
+                        Alternatively this plugin can be enabled/disabled via the `IOPIPE_ENABLE_PROFILER` environment
```
Comment (pselle) [inline iopipe/contrib/profiler/plugin.py]: Environment var update here (to IOPIPE_PROFILER_ENABLE)

File: iopipe/contrib/profiler/plugin.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                        variabl
```
Comment (pselle) [inline iopipe/contrib/profiler/plugin.py]: typo: variabl => variable

**Comment by kolanos** [comment]
@pselle Ready for another review.

File: README.md (hunk 2 above)
Comment (kolanos) [reply README.md]: Changed to `IOPIPE_PROFILER_ENABLED` to be consistent with `IOPIPE_ENABLED`.

**Pull request merged**